wzk/ui/grid/GridColumnsManager.coffee

```diff
@@ -0,0 +1,251 @@
+
+      @checkbs.push checkb
+      goog.events.listen checkb, goog.events.EventType.CLICK, @handleChange
+
+      leaf.appendChild checkb
+      leaf.appendChild label
+
+      list.appendChild leaf
+
+    @saveState @getVisibleCols()
+
+    @removeOld()
+    @el.appendChild list
+
+  ###
+    @protected
+  ###
+  removeOld: =>
+    @dom.removeNode @dom.cls wzk.ui.grid.GridColumnsManager.CLS.LIST, @el
+
+  ###*
+    @protected
+  ###
+  getStorageKey: =>
+    wzk.ui.grid.GridColumnsManager.STORAGE_KEY
+
+  ###*
+    @protected
+  ###
+  getOrCreateState: =>
+    cols = (`/** @type Object */`) @lStorage.get @sKey
+    if @cols.length is 0
+      for col in @allCols
+        cols[col] = true
+      @saveState cols
+
+    cols
+
+  ###*
+    @protected
+  ###
+  getInitialFilterStates: =>
+    states = {}
+    for col in @allCols
+      states[col] = false
+    states
+
+  ###*
+    @protected
+  ###
+  createStorage: =>
+    @lStorage.set @sKey, {} if @lStorage.get(@sKey) is null
+
+  ###*
+    @protected
+    @return {Object}
+  ###
+  getVerboseColNames: =>
+    verboseNames = {}
+    for span, i in @dom.all 'th > span', @table
+      verboseNames[@allCols[i]] = @dom.getTextContent span
+    verboseNames
+
+  ###*
+    @protected
+  ###
+  handleChange: =>
+    @cols = @getVisibleCols()
+    @show @cols
+    @saveState @cols
+
+  ###*
+    @protected
+  ###
+  filterChanged: =>
+    query = @filterWatcher.getQuery()
+    @setFilterStates query
+    @render()
+
+  ###*
+    @protected
+    @param {wzk.resource.Query} query
+  ###
+  setFilterStates: (query) =>
+    for name, isFiltered of @filterStates
+      @filterStates[name] = query.filters[name]?
+
+  ###*
+    @protected
+    @param {Object} state
+  ###
+  saveState: (state) =>
+    @updateFilteringText state
+    @lStorage.set @sKey, state
+
+  ###*
+    @return {Object}
+  ###
+  getVisibleCols: =>
+    cols = {}
+    for checkb, i in @checkbs
+      cols[checkb.name] = checkb['checked']? and checkb['checked']
+    cols
+
+  ###*
+    @protected
+    @param {Object} state
+  ###
+  updateFilteringText: (state) =>
+    hiddenFiltered = (@verboseCols[colName].toLowerCase() for colName, isVisible of state when not isVisible and @filterWatcher.getQuery().filters[colName]?)
+    if hiddenFiltered.length > 0
+      @filteringTextEl.innerHTML = @filteringText.replace('%(columns)s', hiddenFiltered.join(', '))
+      @dom.show @filteringTextEl
+    else
+      @dom.hide @filteringTextEl
+
+  ###*
+    @protected
+  ###
+  showOrHideHeader: =>
+    for checkb, i in @checkbs
+      if @cols[checkb['name']]
+        @showHeader i
+      else
+        @hideHeader i
+
+  ###*
+    @protected
+    @param {number} index
+  ###
+  hideHeader: (index) =>
+    @dom.hide @headers[index] if @headers[index]?
+
+  ###*
+    @protected
+    @param {number} index
+  ###
+  showHeader: (index) =>
+    @dom.show @headers[index] if @headers[index]?
```

Comment:
Dont think that it will do the trick.
http://docs.closure-library.googlecode.com/git/namespace_goog_style.html
Second parameter is used as "isShown", however here I need to check whether index is in the @headers array